spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py

```diff
@@ -237,14 +251,11 @@ def test_get_calltips(ipyconsole, qtbot, function, signature, documentation):
         qtbot.keyClicks(control, 'np.' + function + '(')
 
     # Wait a little bit for the calltip to appear
-    qtbot.wait(500)
-
-    # Assert we displayed a calltip
-    assert control.calltip_widget.isVisible()
```

Comment:
I don't like too much changing `assert` statements by `qtbot.waitUntil` ones because the latter ones are not really tests.

@goanpeca, what do you think?

Reply:
I agree with @ccordoba12. We should tests with assert in generals and use `waitUntil` to replace `qtbot.wait` that are generic and might take longer that waitUntil

Reply:
I see your point, but here I am not replacing just an `assert`, I am replacing a `wait` + `assert`.
I mean the trade-off is: Faster tests vs slightly more explicit tests?

Reply:
My point of view is:
 - If something shouldn't change now => use `assert`
 - If something should change now => use `assert`
 - If something shouldn't change in the future => use `wait` + `assert`
 - If something should change in the future => use `waitUntil` to continue the test as quickly as possible

Reply:
You can also use `waitUntil` and `assert`, this is verbose but fast and explicit.

Reply:
If I write:

```
qtbot.waitUntil(lambda: condition)
assert condition
```
Then `assert condition` can not fail (Because it has to be `True` to get out of the `waitUntil`)

Reply:
@impact27 the main point is that explicit tests should use assert and not waituntil. If we want to speed up tests to wait for a specific thing we can use waitUntil but we have to use assert to make it explicit.

If the condition is to wait for some text (any text) then we should check if the context is text and then assert the actual exact text we are waiting.

>Then assert condition can not fail

Maybe, but it makes things explicit. Again waitUntil should be used to control flow of the runnning test, and not to make the actual assertions.